### ISSUE TITLE
Support Remote Repositories

### ISSUE THREAD

**Issue by roblourens** [microsoft/vscode-livepreview#489]
<img width="664" alt="image" src="https://user-images.githubusercontent.com/323878/228994686-42cdfb49-86c4-4ac5-8087-2ea6c5d8cd9f.png">

It seems like this should be possible in vscode desktop- I assume that somewhere the http server reads a file from disk and serves it, and if it did this using the `vscode.fs` module then this would work? Or is there something that you need native filesystem access for?

**Comment by roblourens** [comment]
The reason I'm looking for this is that I want a workflow for someone else where they can work on a github repo with html files without installing git

**Comment by andreamah** [comment]
The server actually uses the native node `fs` library. Are there any downsides to using the `vscode.fs` API over the node one? It's mostly just the ability to host a server `http.createServer` and the filesystem reads that made me disable this on vscode desktop.

**Comment by andreamah** [comment]
Looking more into this, are you talking about the `workspace.fs` API? I read the injected script content from the extension directory, so I'd need to do a file read outside of the workspace.

**Comment by andreamah** [comment]
Oh wait, I see here that there's a full API for filesystem https://code.visualstudio.com/api/references/vscode-api#FileSystem.

**Comment by roblourens** [comment]
Yeah, and in vscode desktop you should still be able to use the node fs API to read files out of the extension dir if you have to, you would just use the vscode fs API for the content that comes from the user's workspace.

**Comment by TylerLeonhardt** [comment]
https://marketplace.visualstudio.com/items?itemName=codespaces-Contrib.codeswing offers something similar right now

**Comment by roblourens** [comment]
Oh sweet, that proves that the scenario is even possible in web. It injects this into my html page, and then the relative references to css files work

<img width="915" alt="image" src="https://user-images.githubusercontent.com/323878/229006428-78f3d3c9-0188-40d2-a3cb-48118f07b999.png">

However I can't really make it do the exact workflow that I want- it's not designed for the "just open an existing html file" scenario. But I can create a fresh codeswing and paste in my html and then that just works.

**Comment by roblourens** [comment]
I think all it takes is registering a filesystem provider for this custom scheme https://github.com/lostintangent/codeswing/blob/09afee082fb520f2d8746c0f19d6ae2c2565a82f/src/preview/proxyFileSystemProvider.ts#L106

and then set the base to a URI with their custom scheme that is computed here https://github.com/lostintangent/codeswing/blob/09afee082fb520f2d8746c0f19d6ae2c2565a82f/src/preview/webview.ts#L290
